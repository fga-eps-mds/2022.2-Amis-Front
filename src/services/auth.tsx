import { ErrorResponse } from "@remix-run/router";
import axios from "axios";
import { Roles } from "../context/AuthProvider";
// import api from "./api";

const api = axios.create({
  baseURL: import.meta.env.VITE_AMIS_API_BASE_URL_USER,
});

export async function LoginRequest(
  email: string,
  senha: string,
  role: Roles
): Promise<any> {
  try {
    const params = new URLSearchParams();
    params.append("username", email);
    params.append("password", senha);

    const request = await api.post(`/login/${role}`, params);

    if (request.status === 200) {
      const response = {
        token: request.data.access_token,
      };
      return response;
    }
  } catch (error) {
    // console.log(error);
    // eslint-disable-next-line @typescript-eslint/no-throw-literal
    throw new ErrorResponse(401, "Unauthorized", error);
  }
}

export function setUserLocalStorage(user: any) {
  localStorage.setItem("user", JSON.stringify(user));
}

export function getUserLocalStorage() {
  const user = localStorage.getItem("user");

  if (!user) {
    return null;
  }

  return JSON.parse(user);
}
