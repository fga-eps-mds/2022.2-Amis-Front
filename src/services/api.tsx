/* eslint-disable @typescript-eslint/restrict-template-expressions */
/* eslint-disable no-template-curly-in-string */
import axios from "axios";
import { getUserLocalStorage } from "./auth";

const api = axios.create({
  baseURL: import.meta.env.VITE_AMIS_API_BASE_URL,
});

api.interceptors.request.use(
  (config) => {
    const userLocalStorage = getUserLocalStorage();
    config.headers.Authorization = `Bearer ${userLocalStorage?.token}`;
    return config;
  },
  (err) => {
    //console.log(err);
  }
);

export default api;