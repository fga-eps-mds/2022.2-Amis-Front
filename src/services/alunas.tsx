import { AlunasCadastrarDTO } from "../pages/alunas/dtos/AlunasCadastrar.dto";
import api from "./api";

export const cadastraAluna = async (payload: AlunasCadastrarDTO) => {
  try {
    const response = await api.post("/student/", payload);
    return response;
  } catch (error:any) {
    if (error.response) {
      // Ocorreu um erro com resposta do servidor (por exemplo, status HTTP diferente de 2xx)
      console.log("Erro de resposta do servidor:");
      console.log("Status:", error.response.status);
      console.log("Dados da resposta:", error.response.data);
      console.log("Cabeçalhos da resposta:", error.response.headers);
    } else if (error.request) {
      // A solicitação foi feita, mas não houve resposta do servidor
      console.log("Erro de solicitação:");
      console.log("Erro:", error.request);
    } else {
      // Ocorreu um erro ao configurar a solicitação ou ao processar a resposta
      console.log("Erro durante o processamento da solicitação:");
      console.log("Mensagem de erro:", error.message);
    }
    throw error; // Rejeita a Promise para que o erro seja tratado no chamador
  }
};

export const listarAlunas = async () => {
  
  return await api
    .get("/student/")
    .then((response) => response)
    .catch((error) => error);
};

export const editarAluna = async (
  alunaId: string,
  aluna: Object
) => {
  return await api
    .put("/student/" + alunaId, aluna)
    .then((response) => response)
    .catch((error) => error);
};

export const excluirAluna = async (alunaId: string) => {
  return await api
    .delete("/student/" + alunaId)
    .then((response) => response)
    .catch((error) => error);
};
