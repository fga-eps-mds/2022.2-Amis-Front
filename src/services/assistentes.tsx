import { AssistentesCadastrarDTO } from "../pages/assistentes/dtos/AssistentesCadastrar.dto";
import api from "./api";

export const cadastrarAssistente = async (payload: AssistentesCadastrarDTO) => {
  console.log(payload);

  try {
    const response = await api.post("/socialWorker/", payload);
    return response;
  } catch (error) {
    console.error(error);
    return error;
  }
};

export const listarAssistentes = async () => {
  
  return await api
    .get("/socialWorker/")
    .then((response) => response)
    .catch((error) => error);
};

export const editarAssistente = async (
  assistenteId: string,
  assistente: Object
) => {
  return await api
    .put("/socialWorker/" + assistenteId, assistente)
    .then((response) => response)
    .catch((error) => error);
};

export const excluirAssistente = async (assistenteId: string) => {
  return await api
    .delete("/socialWorker/" + assistenteId)
    .then((response) => response)
    .catch((error) => error);
};
