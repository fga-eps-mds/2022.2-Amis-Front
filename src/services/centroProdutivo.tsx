import { CentrosCadastrarDTO } from "../pages/centroProdutivo/dtos/CentrosCadastrar.dto";
import { apiProduction } from "./api";

export const cadastrarCentro = async (payload: CentrosCadastrarDTO) => {
  return await apiProduction
    .post("/centro/", payload)
    .then((response) => response)
    .catch((error) => error);
};

export const listarCentro = async () => {
  return await apiProduction
    .get("/centro/")
    .then((response) => response)
    .catch((error) => error);
};

export const editarCentro = async (CentroId: string, Centro: Object) => {
  return await apiProduction
    .put("/centro/" + CentroId, Centro)
    .then((response) => response)
    .catch((error) => error);
};

export const excluirCentro = async (CentroId: string) => {
  return await apiProduction
    .delete("/centro/" + CentroId)
    .then((response) => response)
    .catch((error) => error);
};

export const inscreveAlunaCentro = async (
  idCentroProd: string,
  idAluna: string
) => {
  return await apiProduction
    .post(`/centro/${idCentroProd}/inscrever/${idAluna}`)
    .then((response) => response)
    .catch((error) => error);
};
