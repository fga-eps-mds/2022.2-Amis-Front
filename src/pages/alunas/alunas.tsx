import React, { useState } from "react";

import styled from "styled-components";
import Sidebar from "../../shared/components/Sidebar/sidebar";
import Navbarlog from "../../shared/components/NavbarLogada/navbarLogada";
import DataTable from "../../shared/components/TablePagination/tablePagination";
import { GridActionsCellItem, GridRowId } from "@mui/x-data-grid";
import PrimaryButton from "../../shared/components/PrimaryButton/PrimaryButton";
import { queryClient } from "../../services/queryClient";
import { BsFillTrashFill } from "react-icons/bs";
import { AiFillEdit, AiFillEye, AiFillEyeInvisible } from "react-icons/ai";
import { toast } from 'react-toastify';
import { Typography } from "@mui/material";
import ValueMask from "../../shared/components/Masks/ValueMask";

import {
  Box,
  FormControl,
  Button,
  Dialog,
  DialogActions,
  DialogTitle,
  InputAdornment,
  InputLabel,
  MenuItem,
  Modal,
  OutlinedInput,
  Select,
  TextField,
} from "@mui/material";
import { useQuery } from "react-query";
import { useForm, FormProvider } from 'react-hook-form';
import { AlunasListarDTO } from "./dtos/AlunasListar.dto";
import { AlunasCadastrarDTO } from "./dtos/AlunasCadastrar.dto";
import {
  cadastraAluna,
  listaAlunas,
  apagaAluna,
  editaAluna,
} from "../../services/alunas";

const Container = styled.div`
  width: 100%;
  height: 100vh;
  background: ${(props) => props.theme.colors.grey};
  display: inline-flex;
`;

const Content = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
`;

const DivButtons = styled.div`
  width: 85%;
  display: inline-flex;
  justify-content: flex-end;
  gap: 20px;
  margin: 0 auto;
  padding-top: 30px;
`;

const Form = styled.form`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 30px;
`;

const FormText = styled.h1`
  color: #525252;
  font-size: 18px;
  font-weight: 400;
  text-align: left;
  padding-bottom: 25px;
`;

const style = {
  position: "absolute" as const,
  top: "50%",
  left: "50%",
  transform: "translate(-50%, -50%)",
  width: 600,
  bgcolor: "background.paper",
  border: "none",
  boxShadow: 24,
  p: 4,
  padding: "50px",
  height: "85%",
  overflow: "hidden",
  overflowY: "scroll",
};

export function Alunas() {
  const [open, setOpen] = useState(false);
  const [aluna, setAluna] = useState(Object);
  const [showPassword, setShowPassword] = useState(false);
  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
  const [showPasswordError, setShowPasswordError] = useState(false);
  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);
  const [dataTable, setDataTable] = useState<Object[]>([]);
  const [id, setId] = useState<GridRowId>(0);
  const [selectedAluna, setSelectedAluna] = useState(null);
  const [openConfirmation, setOpenConfirmation] = useState(false);
  const [openEdit, setOpenEdit] = useState(false);
  const handleOpenConfirmation = () => setOpenConfirmation(true);
  const handleCloseConfirmation = () => setOpenConfirmation(false);

  const methods = useForm();

  const {
    register,
    trigger,
    handleSubmit,
    watch,
    formState: { errors },
  } = methods;

  function removeSpecialCharacters(string: any) {
    if (typeof string === 'string' || string instanceof String) {
      return string.replace(/[./\-\(\) ]/g, "");
    }
    return "";
  }

  function transformDate(date: any) {
    const parts = date.split('/');
    const transformedDate = `${parts[2]}-${parts[1]}-${parts[0]}`;
    return transformedDate;
  }

  const cadastrarAlunas = async (data: any) => {

    const aluna = {
      nome: data.nome,
      login: data.login,
      cpf: data.cpf,
      telefone: data.telefone,
      data_nascimento: data.data_nascimento,
      senha: data.senha,
      bairro: data.bairro,
      cidade: data.cidade,
      cep: data.cep,
      descricao_endereco: data.descricao_endereco,
      deficiencia: data.deficiencia,
      status: data.status,
      email: data.email,
      idEndereco: 1,
    } as AlunasCadastrarDTO;

    aluna.cpf = removeSpecialCharacters(aluna.cpf);
    aluna.telefone = removeSpecialCharacters(aluna.telefone);
    aluna.cep = removeSpecialCharacters(aluna.cep);
    aluna.data_nascimento=transformDate(aluna.data_nascimento);

    const response = await cadastraAluna(aluna);
    if (response.status === 201) {
      //console.log("Aluna cadastrada com sucesso!")
      handleClose();
      toast.success("Aluna cadastrada com sucesso!");
    } else {
      //console.log("MENSAGEM NEGATIVA!!")
      toast.error("Erro ao cadastrar a aluna.");
    }
    await queryClient.invalidateQueries("listar_alunas");
  };

  const validatePassword = (value: any) => {
    const password = watch("senha"); // Obtém o valor do campo de senha
    if (value === password) {
      return true; // A senha e a confirmação de senha são iguais, a validação é bem-sucedida
    }
    return "As senhas não correspondem"; // A senha e a confirmação de senha não são iguais, a validação falhou
  };

  useQuery("listar_alunas", async () => {
    const response = await listaAlunas();
    const temp: AlunasListarDTO[] = [];
    response.data.forEach((value: AlunasListarDTO, index: number) => {
      temp.push({
        id: index, // Adiciona um id único com base no índice
        login: value.login,
        nome: value.nome,
        cpf: value.cpf,
        data_nascimento: value.data_nascimento,
      });
    });
    setDataTable(temp);
  });


  const deleteAlunas = async () => {
    const selectedAluna = dataTable.find((item) => (item as any).id === id); // Encontra o objeto da aluna com base no ID selecionado
    if (selectedAluna) {
      const login = (selectedAluna as any).login; // Obtém o login da aluna
      const response = await apagaAluna(login); // Passa o login para a função apagaAluna

      if (response.status === 204) {
        toast.success("Aluna excluída com sucesso!");
      } else {
        toast.error("Erro ao excluir a aluna.");
      }

      handleCloseConfirmation();
      await queryClient.invalidateQueries("listar_alunas");
    }
  };


  const editAlunas = async (data: any) => {
    // eslint-disable-next-line array-callback-return
    const aluna = {
      nome: data.nome,
      login: data.login,
      cpf: data.cpf,
      telefone: data.telefone,
      data_nascimento: data.data_nascimento,
      senha: data.senha,
      email: data.email,
      status: data.status,
      idEndereco: 1,
    } as AlunasCadastrarDTO;

    const response = await editaAluna(id.toString(), aluna);
    if (response.status === 200 || response.status === 204) {
      toast.success("Aluna atualizada com sucesso!");
    } else {
      toast.error("Erro na atualização da aluna.");
    }
    setOpenEdit(false);
    await queryClient.invalidateQueries("listar_alunas");
  };

  const columnsTable = [
    { field: "nome", headerName: "Nome", flex: 2 },
    { field: "cpf", headerName: "CPF", flex: 2 },
    { field: "data_nascimento", headerName: "Data Nascimento", flex: 2 },
    {
      field: "actions",
      headerName: "Ações",
      type: "actions",
      flex: 1,
      getActions: (params: { id: GridRowId }) => [
        // eslint-disable-next-line react/jsx-key
        <GridActionsCellItem
          icon={<AiFillEdit size={20} />}
          label="Editar"
          onClick={async () => {
            setId(params.id);
            setOpenEdit(true);
          }}
        />,
        // eslint-disable-next-line react/jsx-key
        <GridActionsCellItem
          icon={<BsFillTrashFill size={18} />}
          label="Deletar"
          onClick={() => {
            setId(params.id);
            const selectedRow = dataTable.find((item) => (item as any).id === params.id);
            if (selectedRow) {
              setSelectedAluna((selectedRow as any).login);
              handleOpenConfirmation();
            }
          }}
        />,
      ],
    },
  ];

  return (
    <Container>
      <Sidebar />
      <Content>
        <Navbarlog text={"Alunas"} />
        <DivButtons>
          <PrimaryButton text={"Cadastrar"} handleClick={handleOpen} />
          {/* <PrimaryButton text={"Editar"} /> */}
        </DivButtons>
        <DataTable data={dataTable} columns={columnsTable} />
        <Dialog
          open={openConfirmation}
          onClose={setOpenConfirmation}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-description"
        >
          <DialogTitle id="alert-dialog-title">
            {`Você tem certeza que deseja excluir a aluna ${selectedAluna}?`}
          </DialogTitle>
          <DialogActions>
            <Button onClick={handleCloseConfirmation}>Não</Button>
            <Button onClick={deleteAlunas} autoFocus>
              Sim
            </Button>
          </DialogActions>
        </Dialog>
      </Content>
      <Modal open={open} onClose={handleClose} aria-label="Modal de Cadastro">
        <Box sx={style}>
          <FormProvider {...methods}>
            <FormText id="cabecalho">
              Preencha corretamente os dados cadastrais.
            </FormText>
            <Form onSubmit={handleSubmit(cadastrarAlunas)}>
              <TextField
                id="outlined-nome"
                required={true}
                label="Nome Completo"
                {...register("nome")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />
              <ValueMask label="cpf" />

              <FormControl sx={{ width: '100%', background: '#F5F4FF' }}>
                <InputLabel id="select-deficiencia-label">Possui deficiência?</InputLabel>
                <Select
                  labelId="select-deficiencia-label"
                  id="select-deficiencia"
                  defaultValue={aluna.deficiencia}
                  {...register('deficiencia')}
                  label="Possui deficiência?"
                >
                  <MenuItem value={true as any}>Sim</MenuItem>
                  <MenuItem value={false as any}>Não</MenuItem>
                </Select>
              </FormControl>

              <ValueMask label="data_nascimento" />

              <ValueMask label="telefone" />

              <TextField
                id="outlined-email"
                label="E-mail"
                {...register("email")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />
              <TextField
                id="outlined-login"
                required={true}
                label="Login"
                {...register("login")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />
              <FormControl
                sx={{ width: "100%", background: "#F5F4FF" }}
                variant="outlined"
              >
                <InputLabel htmlFor="outlined-adornment-password" required={true}>
                  Senha
                </InputLabel>
                <OutlinedInput
                  id="outlined-adornment-password"
                  type={showPassword ? "text" : "password"}
                  {...register("senha")}
                  endAdornment={
                    <InputAdornment position="end">
                      {showPassword ? (
                        <AiFillEyeInvisible
                          aria-label="toggle password visibility"
                          onClick={() => {
                            setShowPassword(!showPassword);
                          }}
                          cursor="pointer"
                          size={20}
                        />
                      ) : (
                        <AiFillEye
                          aria-label="toggle password visibility"
                          onClick={() => {
                            setShowPassword(!showPassword);
                          }}
                          cursor="pointer"
                          size={20}
                        />
                      )}
                    </InputAdornment>
                  }
                  label="Password"
                />
              </FormControl>

              <FormControl
                sx={{ width: "100%", background: "#F5F4FF" }}
                variant="outlined"
              >
                <InputLabel htmlFor="outlined-adornment-confirm-password" required={true}>
                  Confirmar senha
                </InputLabel>
                <OutlinedInput
                  id="outlined-adornment-confirm-password"
                  type={showConfirmPassword ? "text" : "password"}
                  {...register("senha_confirmada", { validate: validatePassword })}
                  endAdornment={
                    <InputAdornment position="end">
                      {showConfirmPassword ? (
                        <AiFillEyeInvisible
                          aria-label="toggle password visibility"
                          onClick={() => {
                            setShowConfirmPassword(!showConfirmPassword);
                          }}
                          cursor="pointer"
                          size={20}
                        />
                      ) : (
                        <AiFillEye
                          aria-label="toggle password visibility"
                          onClick={() => {
                            setShowConfirmPassword(!showConfirmPassword);
                          }}
                          cursor="pointer"
                          size={20}
                        />
                      )}
                    </InputAdornment>
                  }
                  label="Password********"
                />
                {errors.senha_confirmada && (
                  <Typography
                    variant="body2"
                    color="error"
                    sx={{ mt: 0.4, mb: -3 }}
                  >Senha não corresponde!
                  </Typography>
                )}
              </FormControl>
              <TextField
                id="outlined-bairro"
                required={true}
                label="Bairro"
                {...register("bairro")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />

              <TextField
                id="outlined-cidade"
                required={true}
                label="Cidade"
                {...register("cidade")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />

              <TextField
                id="outlined-descricao_endereco"
                required={true}
                label="Descricao Endereco"
                {...register("descricao_endereco")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              />

              <ValueMask label="cep" />

              <FormControl fullWidth>
                <InputLabel id="demo-simple-select-label">
                  Status (Produção, Curso ou Inativo)
                </InputLabel>
                <Select
                  id="simple-select-label-status"
                  labelId="simple-select-status"
                  label="Status(Produção, Curso ou Inativo)"
                  {...register("status")}
                  sx={{ width: "100%", background: "#F5F4FF" }}
                >
                  <MenuItem value={1 as any}>Produção</MenuItem>
                  <MenuItem value={2 as any}>Curso</MenuItem>
                  <MenuItem value={3 as any}>Inativo</MenuItem>
                </Select>
              </FormControl>

              <PrimaryButton text={"Confirmar"} />
            </Form>
          </FormProvider>
        </Box>
      </Modal>

      <Modal open={openEdit} onClose={() => setOpenEdit(false)}>
        <Box sx={style}>
          <FormText>Altere os dados da aluna.</FormText>
          <Form onSubmit={handleSubmit(editAlunas)}>
            <TextField
              id="outlined-nome"
              label="Nome Completo *"
              defaultValue={aluna.nome}
              required={true}
              {...register("nome")}
              sx={{ width: "100%", background: "#F5F4FF" }}
            />
            <TextField
              id="outlined-cpf"
              label="CPF *"
              required={true}
              inputProps={{ maxLength: 11 }}
              defaultValue={aluna.cpf}
              {...register("cpf")}
              sx={{ width: "100%", background: "#F5F4FF" }}
            />
            <TextField
              id="outlined-data_nascimento"
              label="Data de Nascimento"
              defaultValue={aluna.data_nascimento}
              required={true}
              {...register("data_nascimento")}
              sx={{ width: "100%", background: "#F5F4FF" }}
            />

            <TextField
              id="outlined-telefone"
              label="Telefone *"
              defaultValue={aluna.telefone}
              required={true}
              {...register("telefone")}
              sx={{ width: "100%", background: "#F5F4FF" }}
            />
            <TextField
              id="outlined-email"
              label="E-mail"
              {...register("email")}
              sx={{ width: "100%", background: "#F5F4FF" }}
            />
            <TextField
              id="outlined-login"
              label="Login *"
              required={true}
              inputProps={{ maxLength: 120 }}
              defaultValue={aluna.login}
              {...register("login")}
              sx={{ width: "100%", background: "#F5F4FF" }}
            />
            <TextField
              id="outlined-senha"
              label="Senha *"
              defaultValue={aluna.senha}
              required={true}
              {...register("senha")}
              sx={{ width: "100%", background: "#F5F4FF" }}
            />

            <FormControl fullWidth>
              <InputLabel id="demo-simple-select-label">
                Status(Produção, Curso ou Inativo)
              </InputLabel>
              <Select
                id="simple-select-label-status"
                labelId="simple-select-status"
                required={true}
                defaultValue={aluna.status}
                label="Status(Produção, Curso ou Inativo)"
                {...register("status")}
                sx={{ width: "100%", background: "#F5F4FF" }}
              >
                <MenuItem value={1 as any}>Produção</MenuItem>
                <MenuItem value={2 as any}>Curso</MenuItem>
                <MenuItem value={3 as any}>Inativo</MenuItem>
              </Select>
            </FormControl>
            <PrimaryButton text={"Editar"} />
          </Form>
        </Box>
      </Modal>
    </Container>
  );
}
