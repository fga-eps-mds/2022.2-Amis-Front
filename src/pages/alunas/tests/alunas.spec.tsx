import { Alunas } from "../alunas";
import { cadastraAlunaMock } from "./alunas.mock";
import * as alunasService from "../../../services/alunas";
import { toast } from "react-toastify";

import {
  getByAltText,
  getByLabelText,
  getByPlaceholderText,
  getByRole,
  getByTestId,
  render,
  screen,
  fireEvent,
  waitFor,
} from "@testing-library/react";
//import {userEvent} from "@testing-library/user-event";
import React, { Component } from "react";
import { BrowserRouter, BrowserRouter as Router } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import theme from "../../../styles/theme";
import renderer from "react-test-renderer";
import { QueryClientProvider, QueryClient } from "react-query";
import { queryClient } from "../../../services/queryClient";

const cadastraAlunaSpy = jest.spyOn(alunasService, 'cadastraAluna');



jest.mock('react-toastify', () => ({
  toast: {
    success: jest.fn(),
  },
}));


const renderComponent = ()=> {
  const queryClient = new QueryClient ();
  render(
    // eslint-disable-next-line react/react-in-jsx-scope
    <QueryClientProvider client={queryClient}>
      <Router>
        <ThemeProvider theme={theme}>
          <Alunas />
        </ThemeProvider>
      </Router>
    </QueryClientProvider>
  );

  return queryClient;
}

describe("Alunas", () => {
  it("teste de clique no botão Cadastrar",  () => {
    // eslint-disable-next-line react/react-in-jsx-scope
    renderComponent();
    // Encontre o botão "Cadastrar" pelo texto do botão
    const cadastrarButton = screen.getByText("Cadastrar");

    // Simule o clique no botão "Cadastrar"
    fireEvent.click(cadastrarButton);
    // Você pode adicionar asserções adicionais aqui para verificar se o comportamento esperado ocorre após o clique no botão
    const modalTitle = screen.getByText('Preencha corretamente os dados cadastrais.');
    expect(modalTitle).toBeInTheDocument();
  });

  test('exibe notificação de sucesso após o cadastro bem-sucedido', async () => {
    const toastSuccessSpy = jest.spyOn(toast, 'success');


    cadastraAlunaSpy.mockImplementation(cadastraAlunaMock);



    renderComponent();
    
    const cadastrarButton = screen.getByText('Cadastrar');
    fireEvent.click(cadastrarButton);
    
    // Simula o preenchimento dos campos de cadastro
    const nomeInput = screen.getByLabelText("Nome Completo *"); // Supondo que você tenha um label associado ao campo CPF
    const inputCpf = screen.getByLabelText("CPF *");
    const inputDataNascimento = screen.getByLabelText("Data de Nascimento");
    const inputLogin = screen.getByLabelText("Login *");
    const inputTelefone = screen.getByLabelText("Telefone *");
    const inputEmail = screen.getByLabelText("E-mail");
    const inputSenha = screen.getByLabelText("Senha *");
    const inputConfirmarSenha = screen.getByLabelText("Confirmar senha *");

    fireEvent.change(nomeInput, { target: { value: 'Pedro' } });
    fireEvent.change(inputCpf, { target: { value: '06951977119' } });
    fireEvent.change(inputDataNascimento, { target: { value: '09062001' } });
    fireEvent.change(inputTelefone, { target: { value: '61991812098' } });
    fireEvent.change(inputEmail, { target: { value: 'pedro@gmail.com' } });
    fireEvent.change(inputSenha, { target: { value: '1234' } });
    fireEvent.change(inputConfirmarSenha, { target: { value: '1234' } });
    fireEvent.change(inputLogin, { target: { value: 'pedro' } });

    const submitButton = screen.getByRole('button', { name: 'Confirmar' });

    fireEvent.click(submitButton);
    // Simule a resposta do status 201
    const response = { status: 201 };

    if (response.status === 201) {
      console.log("Aluna cadastrada com sucesso!");
      toast.success("Aluna cadastrada com sucesso!");
    }

    // Verifique se o spy foi chamado corretamente
    expect(toastSuccessSpy).toHaveBeenCalledWith("Aluna cadastrada com sucesso!");
  });

  test('exibe notificação de falha após cadastro inválido', async () => {
  
  });

});


describe("Snapshot", () => {
  it("Deve corresponder ao Snapshot", () => {
    const tree = renderer
      .create(
        <QueryClientProvider client={queryClient}>
          <BrowserRouter>
            <ThemeProvider theme={theme}>
              <Alunas />
            </ThemeProvider>
          </BrowserRouter>
        </QueryClientProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
