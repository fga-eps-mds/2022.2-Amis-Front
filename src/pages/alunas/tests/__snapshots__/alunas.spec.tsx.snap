// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Deve corresponder ao Snapshot 1`] = `
<div
  className="sc-ipEyDJ fEkVjI"
>
  <div
    className="sc-bcXHqe kOWTzR"
  >
    <div
      className="sc-hLBbgP jTUZgJ"
    >
      AMIS
    </div>
    <a
      className="sc-gswNZR gptURg"
      href="/"
      onClick={[Function]}
    >
      <svg
        color="#da4d3d"
        fill="currentColor"
        height={22}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "#da4d3d",
          }
        }
        viewBox="0 0 1024 1024"
        width={22}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
        />
      </svg>
      <h1
        className="sc-dkrFOg gvjiCT"
      >
        Tela Inicial
      </h1>
    </a>
    <a
      className="sc-gswNZR jZMjZO"
      href="/alunas"
      onClick={[Function]}
    >
      <svg
        color="#525252"
        fill="currentColor"
        height={22}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "#525252",
          }
        }
        viewBox="0 0 24 24"
        width={22}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3zm9 11v-1a7 7 0 0 0-7-7h-4a7 7 0 0 0-7 7v1h2v-1a5 5 0 0 1 5-5h4a5 5 0 0 1 5 5v1z"
        />
      </svg>
      <h1
        className="sc-dkrFOg gvjiCT"
      >
        Alunas
      </h1>
    </a>
    <a
      className="sc-gswNZR jZMjZO"
      href="/turmas"
      onClick={[Function]}
    >
      <svg
        color="#525252"
        fill="currentColor"
        height={22}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "#525252",
          }
        }
        viewBox="0 0 1024 1024"
        width={22}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M296 250c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H296zm184 144H296c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm-48 458H208V148h560v320c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V108c0-17.7-14.3-32-32-32H168c-17.7 0-32 14.3-32 32v784c0 17.7 14.3 32 32 32h264c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm440-88H728v-36.6c46.3-13.8 80-56.6 80-107.4 0-61.9-50.1-112-112-112s-112 50.1-112 112c0 50.7 33.7 93.6 80 107.4V764H520c-8.8 0-16 7.2-16 16v152c0 8.8 7.2 16 16 16h352c8.8 0 16-7.2 16-16V780c0-8.8-7.2-16-16-16zM646 620c0-27.6 22.4-50 50-50s50 22.4 50 50-22.4 50-50 50-50-22.4-50-50zm180 266H566v-60h260v60z"
        />
      </svg>
      <h1
        className="sc-dkrFOg gvjiCT"
      >
        Turmas
      </h1>
    </a>
    <a
      className="sc-gswNZR jZMjZO"
      href="/assistentes"
      onClick={[Function]}
    >
      <svg
        color="#525252"
        fill="currentColor"
        height={22}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "#525252",
          }
        }
        viewBox="0 0 24 24"
        width={22}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3zm9 11v-1a7 7 0 0 0-7-7h-4a7 7 0 0 0-7 7v1h2v-1a5 5 0 0 1 5-5h4a5 5 0 0 1 5 5v1z"
        />
      </svg>
      <h1
        className="sc-dkrFOg gvjiCT"
      >
        Assistentes
      </h1>
    </a>
    <a
      className="sc-gswNZR jZMjZO"
      href="/sair"
      onClick={[Function]}
    >
      <svg
        color="#525252"
        fill="currentColor"
        height={22}
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "#525252",
          }
        }
        viewBox="0 0 24 24"
        width={22}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 13v-2H7V8l-5 4 5 4v-3z"
        />
        <path
          d="M20 3h-9c-1.103 0-2 .897-2 2v4h2V5h9v14h-9v-4H9v4c0 1.103.897 2 2 2h9c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2z"
        />
      </svg>
      <h1
        className="sc-dkrFOg gvjiCT"
      >
        Sair
      </h1>
    </a>
  </div>
  <div
    className="sc-csuSiG dKtfRl"
  >
    <div
      className="sc-eDvSVe bVlFEU"
    >
      <h1
        className="sc-jSUZER iDPtwI"
      >
         
        Alunas
         
      </h1>
      <div
        className="sc-gKPRtg jPborl"
      >
        <div
          className="sc-iBYQkv deFuDe"
        >
          <h1
            className="sc-ftTHYK dNqcwl"
          >
            JOSÃ‰ MARIA
          </h1>
          <h1
            className="sc-pyfCe eXBbNQ"
          >
            Admin
          </h1>
        </div>
        <div
          className="sc-jrcTuL iLQvuk"
        />
      </div>
    </div>
    <div
      className="sc-eDWCr fOChem"
    >
      <button
        className="sc-kDvujY FpMuN"
        onClick={[Function]}
        type="submit"
      >
        Cadastrar
      </button>
    </div>
    <div
      style={
        {
          "background": "#FFF",
          "height": 420,
          "margin": "0 auto",
          "marginTop": "50px",
          "padding": "0 20px",
          "width": "85%",
        }
      }
    >
      <div
        aria-colcount={4}
        aria-multiselectable={false}
        aria-rowcount={1}
        className="MuiDataGrid-root MuiDataGrid-root--densityStandard css-1v2s6lm-MuiDataGrid-root"
        role="grid"
      >
        <div
          className="MuiDataGrid-main css-204u17-MuiDataGrid-main"
        >
          <div
            className="MuiDataGrid-overlay css-f7tavo-MuiDataGrid-overlay"
          >
            Cannot set properties of null (setting 'scrollLeft')
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
